TomTomSDKExamples/Examples/UseCase/Offline/OfflineNavigationContent.swift

```diff
@@ -0,0 +1,689 @@
+struct OfflineNavigationContent: View {
+    var body: some View {
+        OfflineMainView()
+    }
+}
+
+// MARK: - OfflineNavigationController
+
+/// An observable class to handle flows between modules
+final class OfflineNavigationController: ObservableObject {
+    // MARK: Lifecycle
+
+    convenience init() {
+        let textToSpeech = SystemTextToSpeechEngine()
+        let routePlanner = TomTomSDKRoutePlannerOnline.OnlineRoutePlanner(apiKey: Keys.apiKey)
+        let routeReplanner = DefaultRouteReplanner(routePlanner: routePlanner, replanningPolicy: .findBetter)
+        let locationProvider = DefaultCLLocationProvider()
+        let simulatedLocationProvider = SimulatedLocationProvider(delay: .tt.seconds(1))
+        let navigationConfiguration = NavigationConfiguration(
+            apiKey: Keys.apiKey,
+            locationProvider: simulatedLocationProvider,
+            routeReplanner: routeReplanner,
+            betterProposalAcceptanceMode: .automatic
+        )
+        let navigation = Navigation(configuration: navigationConfiguration)
+        let navigationModel = TomTomSDKNavigationUI.NavigationView.ViewModel(navigation, tts: textToSpeech)
+
+        self.init(
+            locationProvider: locationProvider,
+            simulatedLocationProvider: simulatedLocationProvider,
+            routePlanner: routePlanner,
+            navigation: navigation,
+            navigationModel: navigationModel
+        )
+    }
+
+    init(
+        locationProvider: LocationProvider,
+        simulatedLocationProvider: SimulatedLocationProvider,
+        routePlanner: TomTomSDKRoutePlannerOnline.OnlineRoutePlanner,
+        navigation: TomTomSDKNavigation.Navigation,
+        navigationModel: TomTomSDKNavigationUI.NavigationView.ViewModel
+    ) {
+        self.locationProvider = locationProvider
+        self.simulatedLocationProvider = simulatedLocationProvider
+        self.routePlanner = routePlanner
+        self.navigation = navigation
+        navigationViewModel = navigationModel
+
+        self.navigation.addProgressObserver(self)
+        self.navigation.addRouteObserver(self)
+        locationProvider.start()
+    }
+
+    // MARK: Internal
+
+    let locationProvider: LocationProvider
+    let simulatedLocationProvider: SimulatedLocationProvider
+    let routePlanner: TomTomSDKRoutePlannerOnline.OnlineRoutePlanner
+    let navigation: TomTomSDKNavigation.Navigation
+    let navigationViewModel: TomTomSDKNavigationUI.NavigationView.ViewModel
+
+    let displayedRouteSubject = PassthroughSubject<TomTomSDKRoute.Route?, Never>()
+    let progressOnRouteSubject = PassthroughSubject<Measurement<UnitLength>, Never>()
+    let mapMatchedLocationProvider = PassthroughSubject<LocationProvider, Never>()
+
+    @Published
+    var showNavigationView: Bool = false
+}
+
+// MARK: NavigationProgressObserver
+
+/// Extend the OfflineNavigationController to conform to the NavigationProgressObserver protocol.
```

Comment:
Isn't it clear from the extension `OfflineNavigationController: NavigationProgressObserver` line? Not sure if we need this line. The line below is way more useful.